### DISCUSSION TITLE
Generating ID for each component

### DISCUSSION THREAD

**Discussion by VaZark** [django-components/django-components#195] [Q&A] [answered]
I'm trying to create a set of chart widgets. Django-Components seems to be a perfect for integrating reusable elements like charts for a dashboard. However,  the recommended way to do it with vanilla JS  is to use `getElementById` of a canvas element and pass that reference to the lib.

Is there any function to uniquely identify each instance of a component on a page? 

Currently, it looks the project relies on classes for defining shared behaviour. How do I pass and reference each instance of a component alongside the shared class behaviour? 

A good reference the comes to mind is the `useId` hook on react.

**Comment by VaZark** [comment]
Example component

```html
<canvas class="chart-component" id="myChart"></canvas>
 <!-- myChart needs to unique --> 

<script src="https://cdn.jsdelivr.net/npm/chart.js"></script>

<script>
  const ctx = document.getElementById("myChart"); // ID referred here 
  new Chart(ctx, {
    type: "bar",
    data: {
      labels: ["Red", "Blue", "Yellow", "Green", "Purple", "Orange"],
      datasets: [
        {
          label: "# of Votes",
          data: [12, 19, 3, 5, 2, 3],
          borderWidth: 1,
        },
      ],
    },
    options: {
      scales: {
        y: {
          beginAtZero: true,
        },
      },
    },
  });
</script>
```

**Comment by EmilStenstrom** [comment, marked as answer]
I think the easiest way would be to make the ID a parameter to the component. So you would call it with `{% component 'chart' 'myId' %}`. Would that work for you?

**Reply by VaZark** [reply]
Yup. That's perfectly fine and what I ended up using.

I was just wondering if we could have a standard var/method to identify each instance of the components by passing a randomized string / id in the context by default.

**Reply by EmilStenstrom** [reply]
I'm a little worried that people would want different formats of that randomized id, so it wouldn't be usable in practice. Maybe some of them would use the param as a js variable, other's just need a numeric ID. Feels safer not to guess.
